core/modules/filter/filter.pages.inc

```diff
@@ -77,6 +78,72 @@ function filter_format_editor_link_form($form, &$form_state, $format) {
 }
 
 /**
+ * Autocomplete callback for link textfield.
+ */
+function _filter_link_autocomplete($string) {
+  $matches = array();
+  if ($string) {
+    // Get node matches.
+    $results = db_select('node')
+      ->fields('node', array('nid', 'title'))
+      ->condition('title', db_like($string) . '%', 'LIKE')
+      ->addTag('node_access')
```

Comment:
Should we add a sort order here? Seems like sort based on NID DESC would both be practical and fast. Otherwise MySQL returns based on insert order _usually_, but it's unpredictable based on MySQL version and how the data was imported.